spec.html

```diff
@@ -42053,6 +42053,151 @@ <h1>Set Objects</h1>
+              </td>
+              <td>
+                the Set or similar object.
+              </td>
+            </tr>
+            <tr>
+              <td>
+                [[Size]]
+              </td>
+              <td>
+                a non-negative integer or +∞
+              </td>
+              <td>
+                The reported size of the object.
+              </td>
+            </tr>
+            <tr>
+              <td>
+                [[Has]]
+              </td>
+              <td>
+                a function object
+              </td>
+              <td>
+                The `has` method of the object.
+              </td>
+            </tr>
+            <tr>
+              <td>
+                [[Keys]]
+              </td>
+              <td>
+                a function object
+              </td>
+              <td>
+                The `keys` method of the object.
+              </td>
+            </tr>
+          </table>
+        </emu-table>
+      </emu-clause>
+
+      <emu-clause id="sec-getsetrecord" type="abstract operation">
+        <h1>
+          GetSetRecord (
+            _obj_: an ECMAScript language value,
+          ): either a normal completion containing a Set Record or a throw completion
+        </h1>
+        <dl class="header">
+        </dl>
+        <emu-alg>
+          1. If _obj_ is not an Object, throw a *TypeError* exception.
+          1. Let _rawSize_ be ? Get(_obj_, *"size"*).
+          1. Let _numSize_ be ? ToNumber(_rawSize_).
+          1. NOTE: If _rawSize_ is *undefined*, then _numSize_ will be *NaN*.
+          1. If _numSize_ is *NaN*, throw a *TypeError* exception.
+          1. Let _intSize_ be ! ToIntegerOrInfinity(_numSize_).
+          1. If _intSize_ &lt; 0, throw a *RangeError* exception.
+          1. Let _has_ be ? Get(_obj_, *"has"*).
+          1. If IsCallable(_has_) is *false*, throw a *TypeError* exception.
+          1. Let _keys_ be ? Get(_obj_, *"keys"*).
+          1. If IsCallable(_keys_) is *false*, throw a *TypeError* exception.
+          1. Return a new Set Record { [[Set]]: _obj_, [[Size]]: _intSize_, [[Has]]: _has_, [[Keys]]: _keys_ }.
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-setdatahas" type="abstract operation">
+        <h1>
+          SetDataHas (
+            _setData_: a List of either ECMAScript language values or ~empty~,
+            _value_: an ECMAScript language value,
+          ): a Boolean
+        </h1>
+        <dl class="header">
+        </dl>
+        <emu-alg>
+          1. If SetDataIndex(_setData_, _value_) is ~missing~, return *false*.
+          1. Return *true*.
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-setdataindex" type="abstract operation">
+        <h1>
+          SetDataIndex (
+            _setData_: a List of either ECMAScript language values or ~empty~,
+            _value_: an ECMAScript language value,
+          ): a non-negative integer or ~missing~
```

Comment:
Maybe `~not-found~` instead of `~missing~` now that we're using not-found elsewhere.